doc/designs/edns.md

```diff
@@ -0,0 +1,198 @@
+
+The FreeIPA replica deployment depends on whether the DNS integrated service is deployed, distinguishing between two use cases: with and without DNS integrated service. A replica with DNS Integrated Service will mimic a client configuration: it will use Unbound with a DoT forwarder pointing to the DNS server. A replica without DNS Integrated Service will mimic a server configuration: it will use `Bind` for handling incoming unencrypted queries from localhost and encrypted queries from external sources, along with `Unbound` for outgoing encrypted traffic.
+
+Another important aspect is the client's ability to perform DNS updates (`nsupdate`) whenever its IP address changes, ensuring that its DNS record remains up-to-date. This communication must also be secured using DoT/DoH. Currently, the client doesn't support nsupdate with GSS-TSIG (which relies on GSS-API to obtain the secret TSIG key) and unauthenticated updates. However, yo address this, we integrated `nsupdate` DoT functionality from BIND 9.20 into BIND 9.18, the version currently supported by FreeIPA. As a result, `nsupdate` is now enhanced with new DoT options, allowing it to function correctly in secure environments.
+
+## High-level workflow
+
+Enabling DoT: Administrators enable DoT during initial setup or by updating existing DNS configurations through the CLI.
+
+Certificate Management: FreeIPA generates and assigns TLS certificates to the DNS service. Administrators can also provide custom certificates if needed.
+
+DoT Operations: The DNS components listen for DoT traffic on TCP port 853, providing secure communication to clients.
+
+## How to Use
+
+During the deployment of a FreeIPA server, replica, or client, new options are available to enable DoT support. These options allow administrators to enhance the security of DNS traffic. Here’s how to use the new options:
+
+- Enable DoT: use the `--dns-over-tls` option to enable DoT support during the deployment of clients, servers, or replicas. This option deploys Unbound as a local cache resolver (with /etc/resolv.conf listening on 127.0.0.1) and configures BIND on the server to receive DoT requests. On the client side, only Unbound will be deployed (with /etc/resolv.conf listening on 127.0.0.53). Replica deployment configuration will depend on whether the Integrated DNS service is deployed. If it is, the same server configuration will apply. If the Integrated DNS server is not deployed, the client configuration will apply.
```

Comment:
Clarify who listens where. The second sentence says unbound listens on 127.0.0.1 but doesn't clarify where. The second sentence specifies clients on 127.0.52. So perhaps add "servers" to the first part.
I think that by "Integrated DNS service is deployed" means deployed on the new replica but that isn't clear. It could be interpreted as "deployed somewhere in the topology".

Reply:
fixed